src/gui/DatabaseOpenWidget.cpp

```diff
@@ -301,12 +301,14 @@ QSharedPointer<CompositeKey> DatabaseOpenWidget::buildDatabaseKey()
             m_ui->messageWidget->showMessage(tr("Failed to open key file: %1").arg(errorMsg), MessageWidget::Error);
             return {};
         }
-        if (key->type() != FileKey::Hashed && !config()->get(Config::Messages_NoLegacyKeyFileWarning).toBool()) {
+        if (key->type() != FileKey::KeePass2XMLv2 && key->type() != FileKey::Hashed
+            && !config()->get(Config::Messages_NoLegacyKeyFileWarning).toBool()) {
             QMessageBox legacyWarning;
             legacyWarning.setWindowTitle(tr("Legacy key file format"));
-            legacyWarning.setText(tr("You are using a legacy key file format which may become\n"
-                                     "unsupported in the future.\n\n"
-                                     "Please consider generating a new key file."));
+            legacyWarning.setText(tr("You are using a legacy key file format which may become<br>"
```

Comment:
Given Dominik's feedback, recommend replacing the term "legacy" with "older" or similar.

Personally I think we should remove this message on the database unlock screen and just show it when adding a key to the credentials. At this point everyone who has a legacy key has seen the message already and either doesn't care, ignored it, or changed their key.